### ISSUE TITLE
Implement Heartcore API in Uno

### ISSUE THREAD

**Issue by anthonydotnet** [umbraco/Umbraco.Uno.Issues#61]
I want some programatic control over Uno. Ie. CRUD operations for content, media, members, and even users.

My specific use case right now is that I want to change member passwords, get their emails. I will want to manage media and content in the near future.

It seems that the best way to accomplish this is to implement the Heartcore API in Uno.

**Comment by nul800sebastiaan** [comment]
Given that these are 2 completely separate products, I doubt that this is an option at the moment. I'll have to ask around.

**Comment by anthonydotnet** [comment]
Cool. I would hope that the API is just a hand full of dlls which can be dropped in.

**Comment by sitereactor** [comment]
Not really, but its certainly a small part of it 🙂

**Comment by anthonydotnet** [comment]
All I am interesting in is the API endpoints. Don't need the backoffice stuff from Heartcore.

**Comment by sitereactor** [comment]
Yes, I get that but unfortunately its a little more complicated than that.
Long term it's definitely something we would like to see happen. Maybe even the other way around. But there are a few things that needs to mature, so the timeframe is probably way way in the future.

**Comment by anthonydotnet** [comment]
Are there any other options for controlling Uno programatically, without breaking the automatic upgrades?

Sidecar requires a connection string... would that be available?

**Comment by sitereactor** [comment]
No, unfortunately not at this point.

**Comment by nul800sebastiaan** [comment]
I've put this on the list of ideas for the future, and as Morten said, we don't have immediate plans to add APIs to Uno.
